### DISCUSSION TITLE
Custom registration fields option in the cloud console

### DISCUSSION THREAD

**Discussion by mark-aj-cc** [zitadel/zitadel#4183] [Q&A] [answered]
I can not found the "Custom registration fields" option (as mentioned in #1916) in the cloud console!
Is "Custom registration fields" featured right now?

**Comment by mffap** [comment, marked as answer]
Hi @mark-aj-cc

Can't see "Custom registration fields" mentioned in the Announcement, where do you see this?
Providing custom registration fields linked to user metadata is definitely something for our roadmap.

As a workaround for the moment you would need to provide a custom registration form and create users via our APIs. You could save any custom fields as [metadata](https://docs.zitadel.com/docs/guides/manage/customize/user-metadata) on the user.

**Reply by mffap** [reply]
@hifabienne opened an issue for tracking of this feature: https://github.com/zitadel/zitadel/issues/4185

**Reply by fforootd** [reply]
Hi

The way you can integrate ZITADEL in a custom registration flow work as follows:

1) You[ create a service user](https://docs.zitadel.com/docs/guides/integrate/serviceusers#create-a-service-user) in ZITADEL.
2) You assign the service[ user the management role](https://docs.zitadel.com/docs/guides/integrate/access-zitadel-apis#add-org_owner-to-service-user) "Org User Manager".
3) You create the credentials for the service user (either a [personal access token](https://zitadel.com/blog/new-personal-access-token) or a [private key](https://docs.zitadel.com/docs/guides/integrate/serviceusers#1-generate-a-private-public-key-pair-in-zitadel)).
4) You integrate the[ ZITADEL management API](https://docs.zitadel.com/docs/apis/introduction#management) into your web application. You can find the [API definition from the Rest service here](https://editor.swagger.io/?url=https://zitadel.cloud/openapi/v2/swagger/management.swagger.json)  (the POST /users/human is the one thing you are looking for.
5) You create a tailored GUI to you needs.

What kind of technologies and frameworks are you using? I am happy to help you towards more specific answers.
